### PR TITLE
 Add arcadia module and nodes

### PR DESCRIPTION
**Pull request by lcadzy**

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/CommonConst.java [hunk 1]
```diff
@@ -65,6 +65,7 @@
     public static final String AZURE_SERVERLESS_SPARK_ACCOUNT_ICON_PATH = "AzureServerlessSparkAccount.png";
 
     public static final String SQL_BIG_DATA_CLUSTER_MODULE_ICON_PATH = "SqlBigDataClusterRoot_16x.png";
+    public static final String ARCADIA_WORKSPACE_MODULE_ICON_PATH = "SqlBigDataClusterRoot_16x.png";
```
Comment (lcadzy) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/CommonConst.java]: need new icons

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java [hunk 2]
```diff
@@ -192,6 +202,10 @@ protected void refreshFromAzure() throws AzureCmdException {
                     sparkServerlessClusterRootModule.load(true);
                 }
 
+                if (arcadiaModule != null) {
```
Comment (wezhang) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/AzureModule.java]:
```suggestion
                if (arcadiaModule != null && arcadiaModule.isFeatureEnabled() ) {
```

File: Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkWorkspaceNode.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+public class ArcadiaSparkWorkspaceNode extends RefreshableNode {
+    private static final String ARCADIA_WORKSPACE_ID = ArcadiaSparkWorkspaceNode.class.getName();
+    private static final String ICON_PATH = CommonConst.ClusterIConPath;
+
+    @NotNull
+    private ArcadiaWorkSpace workspace;
+
+    public ArcadiaSparkWorkspaceNode(Node parent, @NotNull ArcadiaWorkSpace workspace) {
+        super(ARCADIA_WORKSPACE_ID, workspace.getName(), parent, ICON_PATH, true);
```
Comment (t-rufang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkWorkspaceNode.java]: Replace `workspace.getName()` with `workspace.getTitleForNode()`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkComputeNode.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+public class ArcadiaSparkComputeNode extends RefreshableNode {
+    private static final String ARCADIA_COMPUTE_ID = ArcadiaSparkComputeNode.class.getName();
+    private static final String ICON_PATH = CommonConst.ClusterIConPath;
+
+    @NotNull
+    private ArcadiaSparkCompute compute;
+
+    public ArcadiaSparkComputeNode(Node parent, @NotNull ArcadiaSparkCompute compute) {
+        super(ARCADIA_COMPUTE_ID, compute.getTitle(), parent, ICON_PATH, true);
```
Comment (t-rufang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkComputeNode.java]: Replace `compute.getTitle()` with `compute.getTitleForNode()`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkClusterRootModuleImpl.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    private static final String ICON_PATH = CommonConst.ARCADIA_WORKSPACE_MODULE_ICON_PATH;
+    private static final String BASE_MODULE_NAME = "Apache Spark on Arcadia";
+
+    public ArcadiaSparkClusterRootModuleImpl(@NotNull Node parent) {
+        super(SERVICE_MODULE_ID, BASE_MODULE_NAME, parent, ICON_PATH, true);
+        this.loadActions();
+    }
+
+    @Override
+    protected void refreshItems() throws AzureCmdException {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkClusterRootModuleImpl.java]: Better with `synchronized`

File: Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkComputeNode.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    private ArcadiaSparkCompute compute;
+
+    public ArcadiaSparkComputeNode(Node parent, @NotNull ArcadiaSparkCompute compute) {
+        super(ARCADIA_COMPUTE_ID, compute.getTitleForNode(), parent, ICON_PATH, true);
+        this.compute = compute;
+        this.loadActions();
+    }
+
+    @Override
+    protected void refreshItems() throws AzureCmdException {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkComputeNode.java]: The same as above.

File: Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkWorkspaceNode.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    private ArcadiaWorkSpace workspace;
+
+    public ArcadiaSparkWorkspaceNode(Node parent, @NotNull ArcadiaWorkSpace workspace) {
+        super(ARCADIA_WORKSPACE_ID, workspace.getTitleForNode(), parent, ICON_PATH, true);
+        this.workspace = workspace;
+        this.loadActions();
+    }
+
+    @Override
+    protected void refreshItems() throws AzureCmdException {
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/arcadia/serverexplore/ArcadiaSparkWorkspaceNode.java]: the same as above.

**Pull request merged**